src/Adapter/Order/CommandHandler/ChangeOrderCurrencyHandler.php

```diff
@@ -82,33 +130,64 @@ public function handle(ChangeOrderCurrencyCommand $command)
             ];
 
             foreach ($fields as $field) {
-                $order_detail->{$field} = Tools::convertPriceFull($order_detail->{$field}, $oldCurrency, $currency);
+                $orderDetail->{$field} = Tools::convertPriceFull($orderDetail->{$field}, $oldCurrency, $newCurrency);
             }
 
-            $order_detail->update();
-            $order_detail->updateTaxAmount($order);
+            $orderDetail->update();
+            $orderDetail->updateTaxAmount($order);
         }
+    }
 
-        $id_order_carrier = (int) $order->getIdOrderCarrier();
+    /**
+     * @param PrestaShopCollection $invoices
+     * @param Currency $oldCurrency
+     * @param Currency $newCurrency
+     */
+    private function updateInvoices(PrestaShopCollection $invoices, Currency $oldCurrency, Currency $newCurrency): void
+    {
+        $fields = $this->getSharedAmountFields();
 
-        if ($id_order_carrier) {
-            $order_carrier = new OrderCarrier((int) $order->getIdOrderCarrier());
-            $order_carrier->shipping_cost_tax_excl = (float) Tools::convertPriceFull(
-                $order_carrier->shipping_cost_tax_excl,
-                $oldCurrency,
-                $currency
-            );
-            $order_carrier->shipping_cost_tax_incl = (float) Tools::convertPriceFull(
-                $order_carrier->shipping_cost_tax_incl,
-                $oldCurrency,
-                $currency
-            );
-            $order_carrier->update();
+        if ($invoices->count()) {
+            foreach ($invoices as $invoice) {
+                foreach ($fields as $field) {
+                    if (isset($invoice->$field)) {
+                        $invoice->{$field} = Tools::convertPriceFull($invoice->{$field}, $oldCurrency, $newCurrency);
+                    }
+                }
+
+                $invoice->save();
+            }
         }
+    }
+
+    /**
+     * @param Order $order
+     * @param Currency $oldCurrency
+     * @param Currency $newCurrency
+     */
+    private function updateOrder(Order $order, Currency $oldCurrency, Currency $newCurrency): void
+    {
+        $fields = $this->getSharedAmountFields();
 
-        // Update order && order_invoice amount
-        // @todo: use private method to handle this
-        $fields = [
+        foreach ($fields as $field) {
```

Comment:
Duplicate from line 153, you maybe can add a method `updateFields($object, $fields, $oldCurrency, $newCurrency)` maybe?

Reply:
https://github.com/PrestaShop/PrestaShop/pull/17672/commits/65c612888ad0037d53868c1661136caf7dc66add implemented your suggestion. Although its bad desing vs duplication (couldn't type the object class well), I think its worth it.